server/configuration.go

```diff
@@ -231,6 +231,84 @@ func (p *Plugin) OnConfigurationChange() error {
 	return nil
 }
 
+// ConfigurationWillBeSaved is invoked before saving the configuration to the
+// backing store.
+// An error can be returned to reject the operation. Additionally, a new
+// config object can be returned to be stored in place of the provided one.
+// Minimum server version: 8.0
+//
+// This demo implementation logs a message to the demo channel whenever config
+// is going to be saved.
+// If the Username config option is set to "invalid" an error will be
+// returned, resulting in the config not getting saved.
+// If the Username config option is set to "replaceme" the config value will be
+// replaced with "replaced".
+func (p *Plugin) ConfigurationWillBeSaved(newCfg *model.Config) (*model.Config, error) {
+	cfg := p.getConfiguration()
+	if cfg.disabled {
+		return nil, nil
+	}
+
+	teams, appErr := p.API.GetTeams()
+	if appErr != nil {
+		p.API.LogError(
+			"Failed to query teams ConfigurationWillBeSaved",
+			"error", appErr.Error(),
+		)
+		return nil, nil
+	}
+
+	msg := "Configuration will be saved"
+
+	configData := newCfg.PluginSettings.Plugins[manifest.Id]
+	js, err := json.Marshal(configData)
+	if err != nil {
+		p.API.LogError(
+			"Failed to marshal config data ConfigurationWillBeSaved",
+			"error", err.Error(),
+		)
+		return nil, nil
+	}
+
+	if err := json.Unmarshal(js, &cfg); err != nil {
```

Comment:
Nit pick: Is is worth allocating another variable here to make it easier to compare new and old config?

Reply:
Sure, although the example doesn't really do any comparison with the previous one.